x-pack/filebeat/input/httpjson/config_auth.go

```diff
@@ -109,14 +111,30 @@ func (o *oAuth2Config) client(ctx context.Context, client *http.Client) (*http.C
 
 	switch o.getProvider() {
 	case oAuth2ProviderAzure, oAuth2ProviderDefault:
-		creds := clientcredentials.Config{
-			ClientID:       o.ClientID,
-			ClientSecret:   o.ClientSecret,
-			TokenURL:       o.getTokenURL(),
-			Scopes:         o.Scopes,
-			EndpointParams: o.getEndpointParams(),
+		if o.User != "" || o.Password != "" {
+			conf := &oauth2.Config{
+				ClientID:     o.ClientID,
+				ClientSecret: o.ClientSecret,
+				Endpoint: oauth2.Endpoint{
+					TokenURL:  o.TokenURL,
+					AuthStyle: 1,
```

Comment:
What does the `AuthStyle: 1` mean? Could we replace the `1` with an appropriate const?

Reply:
AuthStyleAutoDetect AuthStyle = 0
AuthStyleInParams AuthStyle = 1
AuthStyleInHeader AuthStyle = 2

1: AuthStyleInParams sends the "client_id" and "client_secret" in the POST body as application/x-www-form-urlencoded parameters.

[here](https://pkg.go.dev/golang.org/x/oauth2#AuthStyle)